pkg/event/events.go

```diff
@@ -192,104 +235,138 @@ func NewPolicyExceptionEvents(engineResponse engineapi.EngineResponse, ruleResp
 	} else {
 		exceptionMessage = fmt.Sprintf("resource %s was skipped from policy rule %s/%s/%s", resourceKey(engineResponse.PatchedResource), pol.GetNamespace(), pol.GetName(), ruleResp.Name())
 	}
+	regarding := corev1.ObjectReference{
+		// TODO: iirc it's not safe to assume api version is set
+		APIVersion: "kyverno.io/v1",
+		Kind:       pol.GetKind(),
+		Name:       pol.GetName(),
+		Namespace:  pol.GetNamespace(),
+		UID:        pol.GetUID(),
+	}
+	related := engineResponse.GetResourceSpec()
 	policyEvent := Info{
-		Kind:              pol.GetKind(),
-		Name:              pol.GetName(),
-		Namespace:         pol.GetNamespace(),
-		RelatedAPIVersion: engineResponse.PatchedResource.GetAPIVersion(),
-		RelatedKind:       engineResponse.PatchedResource.GetKind(),
-		RelatedName:       engineResponse.PatchedResource.GetName(),
-		RelatedNamespace:  engineResponse.PatchedResource.GetNamespace(),
-		Reason:            PolicySkipped,
-		Message:           policyMessage,
-		Source:            source,
-		Action:            ResourcePassed,
+		Regarding: regarding,
+		Related: &corev1.ObjectReference{
+			APIVersion: related.APIVersion,
+			Kind:       related.Kind,
+			Name:       related.Name,
+			Namespace:  related.Namespace,
+			UID:        types.UID(related.UID),
+		},
+		Reason:  PolicySkipped,
+		Message: policyMessage,
+		Source:  source,
+		Action:  ResourcePassed,
 	}
 	exceptionEvent := Info{
-		Kind:              "PolicyException",
-		Name:              exceptionName,
-		Namespace:         exceptionNamespace,
-		RelatedAPIVersion: engineResponse.PatchedResource.GetAPIVersion(),
-		RelatedKind:       engineResponse.PatchedResource.GetKind(),
-		RelatedName:       engineResponse.PatchedResource.GetName(),
-		RelatedNamespace:  engineResponse.PatchedResource.GetNamespace(),
-		Reason:            PolicySkipped,
-		Message:           exceptionMessage,
-		Source:            source,
-		Action:            ResourcePassed,
+		Regarding: corev1.ObjectReference{
+			// TODO: iirc it's not safe to assume api version is set
+			APIVersion: "kyverno.io/v2",
```

Comment:
Started to think why not extract these versions to constants?

Reply:
We definitely could, to me the version doesn't seem relevant in this case anyway.